app/src/main/java/com/example/footprnt/Profile/EditPost.java

```diff
@@ -50,36 +51,46 @@ protected void onCreate(Bundle savedInstanceState) {
         Intent intent = this.getIntent();
         Bundle bundle = intent.getExtras();
         final Post post = (Post) bundle.getSerializable(Post.class.getSimpleName());
+        final int position = (int) bundle.getSerializable("position");
+
         // Set views
         mIvPicture = findViewById(R.id.ivPicture);
         mTvDate = findViewById(R.id.tvPostedAt);
         mEtDescription = findViewById(R.id.etDescription);
         mEtTitle = findViewById(R.id.etTitle);
         mEtLocation = findViewById(R.id.etLocation);
         mBtnDelete = findViewById(R.id.btnDelete);
+        mBtnSave = findViewById(R.id.btnSave);
 
         String date = util.getRelativeTimeAgo(post.getCreatedAt().toString());
         mTvDate.setText(date);
         mEtTitle.setText(post.getTitle());
         mEtDescription.setText(post.getDescription());
-        String city = post.getCity();
-        String country = post.getCountry();
-        String continent = post.getContinent();
-        mEtLocation.setText(city + ", " + country + ", " + continent);
 
-        //TODO: Add save/delete button
+        StringBuilder sb = new StringBuilder();
+        if(post.getCity()!=null){
+            sb.append(post.getCity()+",");
+        }
+        if(post.getCountry()!=null){
+            sb.append(post.getCountry()+",");
+        }
+
+        if(post.getContinent()!=null){
+            sb.append(post.getContinent());
+        }
+
+        mEtLocation.setText(sb);
+
         if (post.getImage() != null) {
             Glide.with(this).load(post.getImage().getUrl()).into(mIvPicture);
-        } else {
-            Glide.with(this).load(R.drawable.ic_add_photo).into(mIvPicture);
         }
 
         // Set view window to be smaller for pop up effect
         DisplayMetrics displayMetrics = new DisplayMetrics();
         getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
         int width = displayMetrics.widthPixels;
         int height = displayMetrics.heightPixels;
-        getWindow().setLayout((int) (width * .8), (int) (height * .7));
+        getWindow().setLayout((int) (width * .8), (int) (height * .8));
```

Comment:
Digging through this class to find the .7 to change to a .8 is the reason why I push you to make these class constants for easier configuration as you test or change your mind.